src/Component/RuleTable/RuleTable.tsx

```diff
@@ -96,61 +93,45 @@ export interface RuleRecord extends GsRule {
   minScale?: number;
 }
 
-// default props
-interface RuleTableDefaultProps extends Partial<TableProps<RuleRecord>> {
+export interface RuleTableProps {
   /** Locale object containing translated text snippets */
-  locale: GeoStylerLocale['RuleTable'];
+  locale?: GeoStylerLocale['RuleTable'];
   /** The renderer to use */
-  rendererType: 'SLD' | 'OpenLayers';
-}
-
-// non default props
-export interface RuleTableProps extends Partial<RuleTableDefaultProps> {
+  rendererType?: 'SLD' | 'OpenLayers';
   /** Reference to internal data object (holding schema and example features) */
   data?: Data;
   /** List of rules to display in rule table */
   rules: GsRule[];
-  /** Properties of the SLD renderer */
-  sldRendererProps?: SLDRendererAdditonalProps;
-  /** Properties of the OpenLayers renderer */
-  oLRendererProps?: Partial<OlRendererProps>;
   /** The footer of the rule table */
   footer?: (currentPageData?: any) => React.ReactNode;
   /** The callback function that is triggered when the rules change */
   onRulesChange?: (rules: GsRule[]) => void;
   /** The callback function that is triggered when the selection changes */
   onSelectionChange?: (selectedRowKeys: string[], selectedRows: any[]) => void;
   /** Properties that will be passed to the Comparison Filters */
-  filterUiProps?: Partial<ComparisonFilterProps>;
-  /** List of supported icons ordered as library */
-  iconLibraries?: IconLibrary[];
-  /** Object containing predefined color ramps */
-  colorRamps?: {
-    [name: string]: string[];
-  };
 }
 
 const COMPONENTNAME = 'RuleTable';
 
 // export class RuleTable extends React.Component<RuleTableProps, RuleTableState> {
-export const RuleTable: React.FC<RuleTableProps> = (props) => {
-
-  const composition = useGeoStylerComposition('Rule', {});
+export const RuleTable: React.FC<RuleTableProps & RuleComposableProps & TableProps<RuleRecord>> = (props) => {
```

Comment:
Is there a reasoning when to use the `interal` props approach and when not?

Reply:
Just forgot it here. :)